ddl/ddl_tiflash_api.go

```diff
@@ -397,53 +535,44 @@ func (d *ddl) pollTiFlashReplicaStatus(ctx sessionctx.Context, pollTiFlashContex
 		})
 		// We only check unavailable tables here, so doesn't include blocked add partition case.
 		if !available {
-			allReplicaReady = false
 			enabled, inqueue, _ := pollTiFlashContext.Backoff.Tick(tb.ID)
 			if inqueue && !enabled {
 				logutil.BgLogger().Info("Escape checking available status due to backoff", zap.Int64("tableId", tb.ID))
 				continue
 			}
 
-			// We don't need to set accelerate schedule for this table, since it is already done in DDL, when
-			// 1. Add partition
-			// 2. Set TiFlash replica
-
-			// Compute sync data process by request TiFlash.
-			regionReplica := make(map[int64]int)
-			for _, store := range pollTiFlashContext.TiFlashStores {
-				err := helper.CollectTiFlashStatus(store.Store.StatusAddress, tb.ID, &regionReplica)
+			progress, err := getTiFlashTableSyncProgress(pollTiFlashContext, tb.ID, tb.Count)
+			if err != nil {
+				logutil.BgLogger().Error("get tiflash sync progress failed",
+					zap.Error(err),
+					zap.Int64("tableID", tb.ID),
+				)
+				continue
+			}
+			if pollTiFlashContext.ProgressCache[tb.ID] != progress {
+				err = infosync.UpdateTiFlashTableSyncProgress(context.Background(), tb.ID, progress)
```

Comment:
Will it cause problem during upgrading? Will it happen that some old TiDB trying to parse old format, or the opposite case?